File: frontend/src/services/StatusService.js
import axios from "axios";

export default class StatusService{
    getAll() {
        return axios.get("http://localhost:8080/api/status/getAll")
    }

    getById(id) {
        return axios.get(`http://localhost:8080/api/status/getById?id=${id}`)
    }

    getByName(name){
        return axios.get(`http://localhost:8080/api/status/getByName?name=${name}`)
    }
}

File: backend/src/main/java/ecrow/backend/business/abstracts/StatusService.java
package ecrow.backend.business.abstracts;

import ecrow.backend.core.utilities.results.DataResult;
import ecrow.backend.entities.concretes.Status;

import java.util.List;

public interface StatusService {
    DataResult<List<Status>> getAll();
    DataResult<Status> getById(Integer id);
    DataResult<Status> getByName(String name);
}
